test/test_mps.py

```diff
@@ -6985,18 +9711,107 @@ def test_output_match(self, device, dtype, op):
                 cpu_out = op(*cpu_args, **cpu_kwargs)
                 mps_out = op(*mps_args, **mps_kwargs)
                 self.assertEqual(cpu_out, mps_out)
-        except Exception as e:
-            if not generate_new_truth:
-                raise e
-        else:
-            if generate_new_truth:
-                self.NEW_ALLOW_LIST[op.name].append(str(dtype))
 
-                # We could write it only once. But I don't know how to detect that the current test is the last one
-                # So each test append to the dict and write it.
+            except Exception as e:
+                if not generate_new_truth:
+                    raise e
+                forward_failed = True
+                all_forward_pass = False
+
+            # Skip the grad test if it is not part of the allow list
+            if not generate_new_truth and not run_grad_test:
+                if i == 0:
+                    print(
+                        f"Skipping gradient check because {op.name} is not on the allow list"
+                    )
+                continue
+
+            if not (dtype.is_floating_point or dtype.is_complex):
+                continue
```

Comment:
This means that you never generate forward new truth for non-floating/complex dtypes?